### PR TITLE
feat: improve lifecycle messages

### PR DESCRIPTION
**Pull request by sergiusens**
- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `make lint`?
- [ ] Have you successfully run `pytest tests/unit`?

-----

### CODE REVIEW

**Review by cmatsuoka — APPROVED** [review]
It would be nice to have similar messages in craft-application as well to make the transition smoother.

**Comment by sergiusens** [comment]
@cmatsuoka I copied them from craft application!

File: snapcraft/parts/parts.py [hunk 1]
```diff
@@ -185,11 +185,10 @@ def run(
                             project_vars=action.project_vars,
                             properties=action.properties,
                         )
-                    message = _action_message(action)
-                    emit.progress(f"Executing parts lifecycle: {message}")
-                    with emit.open_stream("Executing action") as stream:
+                    message = _get_parts_action_message(action)
+                    emit.progress(message)
+                    with emit.open_stream(message) as stream:
```
Comment (tigarmo) [inline snapcraft/parts/parts.py:190]: Suggestion: bump craft-cli to 2.4.0 and then remove this parameter

Comment (sergiusens) [reply snapcraft/parts/parts.py:188]: done

**Review by mr-cal — APPROVED** [review]

**Pull request merged**